charms/istio-ingressgateway/src/charm.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python3
+
+import logging
+from pathlib import Path
+import os
+from kubernetes import client, config
+
+from ops.charm import CharmBase
+from ops.main import main
+from ops.model import ActiveStatus, MaintenanceStatus, WaitingStatus, BlockedStatus
+
+from oci_image import OCIImageResource, OCIImageResourceError
+from k8s_service import RequireK8sService
+
+
+class IstioIngressGatewayCharm(CharmBase):
+    def __init__(self, *args):
+        super().__init__(*args)
+        if not self.unit.is_leader():
+            # We can't do anything useful when not the leader, so do nothing.
+            self.model.unit.status = WaitingStatus("Waiting for leadership")
+            return
+        self.log = logging.getLogger(__name__)
+        self.istio_pilot = RequireK8sService(self, "istio-pilot")
+        self.image = OCIImageResource(self, "oci-image")
+        for event in [
+            self.on.install,
+            self.on.leader_elected,
+            self.on.upgrade_charm,
+            self.on.update_status,
+            self.on.config_changed,
+            self.istio_pilot.on.k8s_services_changed,
+        ]:
+            self.framework.observe(event, self.main)
+
+    def main(self, event):
+        try:
+            image_details = self.image.fetch()
+        except OCIImageResourceError as e:
+            self.model.unit.status = e.status
+            self.log.info(e)
+            return
+
+        self.model.unit.status = MaintenanceStatus("Setting pod spec")
+
+        if not self.istio_pilot.is_created:
+            self.model.unit.status = BlockedStatus("Needing Istio Pilot relation")
+            return
+
+        if not self.istio_pilot.is_available:
+            self.model.unit.status = WaitingStatus("Waiting for Istio Pilot information")
+            return
+
+        pilot_service, pilot_port = self.istio_pilot.services[0]
+        cfg = self.model.config
+        namespace = self.model.name
+        service_name = self.model.app.name
+
+        config_map = self.check_ca_root_cert(namespace)
```

Comment:
Can this get combined with the `if not self.istio_pilot.is_available` check above? It seems like the `Waiting for Istio Pilot information` status message is also appropriate for this check.